samples/lucy_utils.cpp

```diff
@@ -0,0 +1,365 @@
+            .targetApi(filamat::MaterialBuilder::TargetApi::OPENGL)
+            .shading(Shading::UNLIT)
+            .depthWrite(false)
+            .depthCulling(false)
+            .build();
+        return Material::Builder().package(pkg.getData(), pkg.getSize()).build(engine);
+    }(*engine);
+
+    TextureSampler::MinFilter minFilter;
+    TextureSampler::MagFilter magFilter;
+    MaterialInstance* matInstance;
+    switch (op) {
+        case BLIT:
+            minFilter = TextureSampler::MinFilter::LINEAR;
+            magFilter = TextureSampler::MagFilter::LINEAR;
+            matInstance = blit->createInstance();
+            break;
+        case HBLUR:
+            minFilter = TextureSampler::MinFilter::NEAREST;
+            magFilter = TextureSampler::MagFilter::NEAREST;
+            matInstance = blur->createInstance();
+            matInstance->setParameter("weights", hweights, FILTER_SIZE);
+            break;
+        case VBLUR:
+            minFilter = TextureSampler::MinFilter::NEAREST;
+            magFilter = TextureSampler::MagFilter::NEAREST;
+            matInstance = blur->createInstance();
+            matInstance->setParameter("weights", vweights, FILTER_SIZE);
+            break;
+        case MIX:
+            minFilter = TextureSampler::MinFilter::LINEAR;
+            magFilter = TextureSampler::MagFilter::LINEAR;
+            matInstance = mix->createInstance();
+            matInstance->setParameter("secondary", secondary, TextureSampler(minFilter, magFilter));
+            break;
+    }
+    matInstance->setParameter("color", tex, TextureSampler(minFilter, magFilter));
+
+    Entity entity = EntityManager::get().create();
+    RenderableManager::Builder(1)
+        .boundingBox({{ 0, 0, 0 }, { 9000, 9000, 9000 }})
+        .material(0, matInstance)
+        .geometry(0, RenderableManager::PrimitiveType::TRIANGLES, vertexBuffer, indexBuffer)
+        .build(*engine, entity);
+
+    return entity;
+}
+
+Entity createDisk(Engine* engine, Texture* reflection) {
+    constexpr int nslices = 64;
+    constexpr int nverts = nslices + 2;
+
+    static float4 verts[nverts];
+    int slice = 0;
+    while (slice < nslices + 1) {
+        float theta = 2.0f * M_PI * slice / nslices;
+        verts[slice++] = float4 {
+            std::cos(theta), std::sin(theta),
+            0.5 + 0.5 * std::cos(theta), 0.5 + 0.5 * std::sin(theta)
+        };
+    }
+    verts[slice] = {0.0f, 0.0f, 0.5f, 0.5f};
+
+    static uint16_t indices[nslices * 3];
+    for (int slice = 0, j = 0; slice < nslices; ++slice) {
+        indices[j++] = nverts - 1;
+        indices[j++] = slice;
+        indices[j++] = (slice + 1) % (nslices + 1);
+    }
+
+    VertexBuffer* vbuffer = VertexBuffer::Builder()
+        .vertexCount(nverts)
+        .bufferCount(1)
+        .attribute(VertexAttribute::POSITION, 0, VertexBuffer::AttributeType::FLOAT2, 0, 16)
+        .attribute(VertexAttribute::UV0, 0, VertexBuffer::AttributeType::FLOAT2, 8, 16)
```

Comment:
Please add tangents here, as the disk renderable is lit (should be a warning in release mode).

Reply:
Done